File: myimages_ui/src/services/ImageRequestService.js
import http from "../http-common"; 

class ImageRequestService {
  getAllImageRequests(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/imageRequest/imageRequests`, searchDTO);
  }
 

  get(imageRequestId) {
    return this.getRequest(`/imageRequest/${imageRequestId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/imageRequest?field=${matchData}`, null);
  }

  addImageRequest(data) {
    return http.post("/imageRequest/addImageRequest", data);
  }

  update(data) {
  	return http.post("/imageRequest/updateImageRequest", data);
  }
  
  uploadImage(data,imageRequestId) {
  	return http.postForm("/imageRequest/uploadImage/"+imageRequestId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ImageRequestService();

File: src/main/java/com/myimages/service/ImageRequestService.java
package com.myimages.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myimages.domain.ImageRequest;
import com.myimages.dto.ImageRequestDTO;
import com.myimages.dto.ImageRequestSearchDTO;
import com.myimages.dto.ImageRequestPageDTO;
import com.myimages.dto.ImageRequestConvertCriteriaDTO;
import com.myimages.service.GenericService;
import com.myimages.dto.common.RequestDTO;
import com.myimages.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ImageRequestService extends GenericService<ImageRequest, Integer> {

	List<ImageRequest> findAll();

	ResultDTO addImageRequest(ImageRequestDTO imageRequestDTO, RequestDTO requestDTO);

	ResultDTO updateImageRequest(ImageRequestDTO imageRequestDTO, RequestDTO requestDTO);

    Page<ImageRequest> getAllImageRequests(Pageable pageable);

    Page<ImageRequest> getAllImageRequests(Specification<ImageRequest> spec, Pageable pageable);

	ResponseEntity<ImageRequestPageDTO> getImageRequests(ImageRequestSearchDTO imageRequestSearchDTO);
	
	List<ImageRequestDTO> convertImageRequestsToImageRequestDTOs(List<ImageRequest> imageRequests, ImageRequestConvertCriteriaDTO convertCriteria);

	ImageRequestDTO getImageRequestDTOById(Integer imageRequestId);


}
